### PR TITLE
Add GitHub action for generating docker image

### PR DESCRIPTION
**Pull request by arturkasperek**

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Publish Docker image
+on:
+  push:
+    tags:
```
Comment (nozwock) [inline .github/workflows/docker.yaml:4]: Seems redundant? when we're already running on push to main

Comment (arturkasperek) [reply .github/workflows/docker.yaml:4]: We need as well catch `tags` to create build when you will make new release
Zrzut ekranu 2023-04-12 o 09 24 16

Comment (nozwock) [reply .github/workflows/docker.yaml:4]: ah is it like for tags on main branch?

Comment (arturkasperek) [reply .github/workflows/docker.yaml:4]: Zrzut ekranu 2023-04-12 o 09 27 10
it's for above tags

Comment (nozwock) [reply .github/workflows/docker.yaml:4]: Ok so that makes more sense if it's only for tags and not for every commit

Comment (arturkasperek) [reply .github/workflows/docker.yaml:4]: yeah, it's for tags

Comment (nozwock) [reply .github/workflows/docker.yaml:4]: Alright

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+RUN ln -s /usr/bin/g++-12 /usr/bin/g++
+RUN ln -s /usr/bin/gcc-12 /usr/bin/gcc
+RUN ln -s /usr/bin/gcc-12 /usr/bin/cc
+RUN ln -s /usr/bin/gcc-ar-12 /usr/bin/gcc-ar
+RUN ln -s /usr/bin/gcc-nm-12 /usr/bin/gcc-nm
+
+RUN curl https://sh.rustup.rs | sh -s -- -y
+
+COPY . .
+WORKDIR /work/yanu-cli
```
Comment (nozwock) [inline Dockerfile:17]: isn't this supposed to be `/work/crates/yanu-cli`? is this alright?

Comment (arturkasperek) [reply Dockerfile:17]: yeah, makes sense - will change that soon

**Comment by nozwock** [comment]
Also consider renaming package name to `yanu-docker`.
pkg

**Comment by arturkasperek** [comment]
> Also consider renaming package name to `yanu-docker`. pkg

That one doesn't make sense in my opinion cause that information will be redundant

**Comment by nozwock** [comment]
> That one doesn't make sense in my opinion cause that information will be redundant

Well I thought it'd make it more obvious, that it's a docker package without having to look at the. readme

File: Dockerfile (hunk 2 above)
Comment (arturkasperek) [reply Dockerfile]: It's already tweaked by you

**Pull request merged**